flocker/node/functional/test_docker.py

```diff
@@ -170,6 +170,59 @@ def started(_):
         d.addCallback(started)
         return d
 
+    def test_private_registry_image(self):
+        """
+        ``DockerClient.add`` can start containers based on an image from a
+        private registry.
+        """
+        address_user = socket.socket()
+        self.addCleanup(address_user.close)
+        address_user.bind(('', 0))
+        used_address = address_user.getsockname()
+        address_user.close()
+        registry_name = random_name(self)
+        registry_port = used_address[1]
+        registry_starting = self.start_container(
+            unit_name=registry_name,
+            image_name='registry:2',
+            ports=[
+                PortMap(
+                    internal_port=5000,
+                    external_port=registry_port
+                ),
+            ]
+        )
+        repository = '127.0.0.1:{}/clusterhq/testimage'.format(
```

Comment:
> Why?
Because if the removal of the local image fails for some reason, it'll be obvious which test left the image behind and so that subsequent test runs don't use the local image instead of pulling from the temporary repository.

> I think that this is an image_location (or at least, it isn't just a repository)
I agree it's confusing, but repository is the name of the attribute we've used in `flocker.control.DockerImage` and I've re-used that record now.

That record should probably grow separate repository, name and tag attributes which will be part of FLOC-246 now linked from the new test.